src/design-patterns/MazeGame/builder/standard_maze_builder.hpp

```diff
@@ -0,0 +1,30 @@
+#ifndef STANDARD_MAZE_BUILDER_H
+#define STANDARD_MAZE_BUILDER_H
+
+#include <memory>
+
+#include "direction.hpp"
+#include "maze_builder.hpp"
+
+class StandardMazeBuilder : public MazeBuilder {
+ public:
+  StandardMazeBuilder();
+  ~StandardMazeBuilder() = default;
+
+  StandardMazeBuilder(const StandardMazeBuilder&) = delete;
+  StandardMazeBuilder(StandardMazeBuilder&&) = delete;
+  StandardMazeBuilder& operator=(const StandardMazeBuilder&) = delete;
+  StandardMazeBuilder& operator=(StandardMazeBuilder&&) = delete;
+
+  void BuildMaze() override;
+  void BuildRoom(const int room) override;
+  void BuilderDoor(const int roomFrom, const int roomTo) override;
```

Comment:
```suggestion
  void BuildMaze() override final;
  void BuildRoom(const int room) override final;
  void BuilderDoor(const int roomFrom, const int roomTo) override final;
```

아마 StandardMazeBuilder가 더 파생되어서 변경될 일 없으면, 명시해주는게 좋을 듯

Reply:
[modernize-use-override](https://clang.llvm.org/extra/clang-tidy/checks/modernize/use-override.html)

```
'override' is redundant since the function is already declared 'final' (fix available) [modernize-use-override]
```
final만 쓰면 될듯

---

**override:**

- 이 키워드는 하위 클래스에서 상위 클래스의 가상 함수를 오버라이딩한다는 것을 명시적으로 나타냅니다.
- 컴파일러에게 이 함수가 기본 클래스의 가상 함수를 오버라이딩한다는 것을 알려줍니다.
- 이를 통해 실수로 인한 함수 오버라이딩의 오류를 방지할 수 있습니다. 예를 들어, 기본 클래스의 함수 시그니처가 변경되었을 때, override 지정자를 사용한 함수는 컴파일 에러를 발생시켜 개발자에게 알립니다.

**final:**

- 이 키워드는 해당 함수가 더 이상 하위 클래스에서 오버라이딩되지 않아야 함을 나타냅니다.
- 즉, 현재 클래스에서 이 함수의 구현이 "최종"이며, 이후의 상속 계층에서는 변경될 수 없습니다.
- 이는 특정 메서드의 동작을 고정시키고 하위 클래스에서의 변경을 방지하는 데 유용합니다.

### 어느 것을 사용해야 할까요?

- **확장 가능성**: 만약 하위 클래스에서 해당 메서드의 구현을 변경할 수 있게 하고 싶다면 override를 사용하세요.
- **확장 제한**: 만약 이 클래스에서 메서드의 구현을 고정시키고, 이후의 모든 상속에서 이 메서드를 변경하지 못하게 하고 싶다면 final을 사용하세요.